File: variant-z/python-service/app/models/order_line_item.py
"""Order Line Item model (Variant Z)."""

from sqlalchemy import Column, String, DateTime, ForeignKey, Integer, Numeric, func

from app.core.database import Base


class OrderLineItem(Base):
    """Order Line Item model."""
    __tablename__ = 'order_line_items'
    
    id = Column(String(36), primary_key=True)
    order_id = Column(String(36), ForeignKey('orders.id'), nullable=False, index=True)
    sku_id = Column(String(36), ForeignKey('skus.id'), nullable=False, index=True)
    quantity = Column(Integer, nullable=False)
    unit_price = Column(Numeric(10, 2), nullable=False)
    total_price = Column(Numeric(10, 2), nullable=False)
    product_name = Column(String(255), nullable=False)
    sku_code = Column(String(255), nullable=False)
    created_at = Column(DateTime, nullable=False, default=func.now())
    updated_at = Column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

File: variant-z/java-service/src/main/java/com/flashsale/entity/OrderLineItem.java
package com.flashsale.entity;

import jakarta.persistence.*;

/**
 * Order line item entity - Variant Z.
 */
@Entity
@Table(name = "order_line_items")
public class OrderLineItem {

    @Id
    @Column(name = "id", updatable = false, nullable = false)
    private String id;

    @Column(name = "order_id", nullable = false)
    private String orderId;

    @Column(name = "sku_id", nullable = false)
    private String skuId;

    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    @Column(name = "unit_price", nullable = false)
    private Double unitPrice;

    @Column(name = "total_price", nullable = false)
    private Double totalPrice;

    @Column(name = "product_name", nullable = false)
    private String productName;

    @Column(name = "sku_code", nullable = false)
    private String skuCode;

    // Default constructor
    public OrderLineItem() {}

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getOrderId() { return orderId; }
    public void setOrderId(String orderId) { this.orderId = orderId; }
    public String getSkuId() { return skuId; }
    public void setSkuId(String skuId) { this.skuId = skuId; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public Double getUnitPrice() { return unitPrice; }
    public void setUnitPrice(Double unitPrice) { this.unitPrice = unitPrice; }
    public Double getTotalPrice() { return totalPrice; }
    public void setTotalPrice(Double totalPrice) { this.totalPrice = totalPrice; }
    public String getProductName() { return productName; }
    public void setProductName(String productName) { this.productName = productName; }
    public String getSkuCode() { return skuCode; }
    public void setSkuCode(String skuCode) { this.skuCode = skuCode; }
}